app/models/comment.rb

```diff
@@ -3,6 +3,22 @@ class Comment < ApplicationRecord
   belongs_to :commentable, :polymorphic => true
   validates :body, presence: true
 
+
+  def source_question
+    if commentable_type == "Question"
+      commentable
+    else
+      commentable.question
+    end
+  end
+
+  def self.comments_to_user_activity(user_id)
```

Comment:
Whew! Look at those left outer joins. I really like the decisions you made about this.